packages/jest-core/src/SearchSource.ts

```diff
@@ -50,6 +50,15 @@ const toTests = (context: Context, tests: Array<Config.Path>) =>
     path,
   }));
 
+const hasSCM = (changedFilesInfo: ChangedFiles) => {
+  const {repos} = changedFilesInfo;
+  // no SCM (git/hg/...) is found in any of the roots.
+  const noSCM = (Object.keys(repos) as Array<
+    keyof ChangedFiles['repos']
+  >).every(scm => repos[scm].size === 0);
```

Comment:
```suggestion
  const noSCM = Object.values(repos).every(scm => scm.size === 0);
```

might need the type there